yt_dlp/extractor/netverse.py

```diff
@@ -0,0 +1,174 @@
+from .common import InfoExtractor
+from ..utils import (
+    traverse_obj,
+    ExtractorError,
+    # GeoRestrictedError,
+)
+
+
+class NetverseBaseIE(InfoExtractor):
+    def get_required_json(self, url, data=None):
+        match = self._match_valid_url(url).groupdict()
+        display_id, sites_type = match['display_id'], match['type']
+
+        if sites_type == 'watch' or sites_type == 'video':
+            media_api_url = f'https://api.netverse.id/medias/api/v2/watchvideo/{display_id}'
+        elif sites_type == 'webseries':
+            media_api_url = f'https://api.netverse.id/medias/api/v2/webseries/{display_id}'
+
+        json_data = self._download_json(media_api_url, display_id, data=data)
+
+        if json_data.get('error'):
+            raise ExtractorError(json_data.get('message'))
+
+        return json_data
+
+    def get_access_id(self, dailymotion_url):
+        return dailymotion_url.split('/')[-1]
```

Comment:
The url may contain query. Use `urllib.parse`

Also, does this even need to be a function?